### PR TITLE
Implement the draft of the dynamic adjustment of weights

### PR DESCRIPTION
**Pull request by echozzy629**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,19 +13,24 @@
...
   "dependencies": {
     "@dagrejs/dagre": "^1.0.4",
+    "@mui/lab": "^5.0.0-alpha.162",
+    "@mui/material": "^5.15.6",
     "@radix-ui/react-accordion": "^1.1.2",
     "@radix-ui/react-icons": "^1.3.0",
     "@radix-ui/react-separator": "^1.0.3",
     "@radix-ui/react-slider": "^1.1.2",
     "@radix-ui/themes": "^2.0.1",
+    "@react-three/fiber": "^8.15.14",
```
Comment (devjeetr) [inline package.json]: are we using this?

File: package.json [hunk 2]
```diff
@@ -13,19 +13,24 @@
   },
   "dependencies": {
     "@dagrejs/dagre": "^1.0.4",
+    "@mui/lab": "^5.0.0-alpha.162",
+    "@mui/material": "^5.15.6",
```
Comment (devjeetr) [inline package.json]: these as well

File: package.json [hunk 3]
```diff
@@ -13,19 +13,24 @@
...
     "@stitches/react": "^1.2.8",
     "d3": "^7.8.5",
     "jotai": "^2.5.1",
     "ramda": "^0.29.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-select": "^5.8.0",
     "reactflow": "^11.10.1",
     "recharts": "^2.10.4",
+    "three": "^0.160.1",
```
Comment (devjeetr) [inline package.json]: are we using this?

File: package.json [hunk 4]
```diff
@@ -13,19 +13,24 @@
...
     "@radix-ui/react-slider": "^1.1.2",
     "@radix-ui/themes": "^2.0.1",
+    "@react-three/fiber": "^8.15.14",
     "@stitches/react": "^1.2.8",
     "d3": "^7.8.5",
     "jotai": "^2.5.1",
     "ramda": "^0.29.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-select": "^5.8.0",
```
Comment (devjeetr) [inline package.json]: are we using this?

**Review by devjeetr — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (echozzy629) [reply package.json]: removed

File: package.json (hunk 3 above)
Comment (echozzy629) [reply package.json]: removed

File: package.json (hunk 2 above)
Comment (echozzy629) [reply package.json]: removed

File: package.json (hunk 1 above)
Comment (echozzy629) [reply package.json]: removed

**Pull request merged**